### PR TITLE
Build Docker dev image

### PR DESCRIPTION
**Pull request by pamolloy**

### CODE REVIEW

**Comment by pamolloy** [comment]
Once I actually get DepthAI running in a container it may require updating this, but it is perfectly functional now as a base image.

File: .github/workflows/docker-hub.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    branches:
+      - master
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up QEMU
```
Comment (themarpe) [inline .github/workflows/docker-hub.yml:13]: Is QEMU needed?

**Review by themarpe — CHANGES_REQUESTED** [review]
I don't know whether this is the right location for this, WRT having other docker images as well.  
And another thing to discuss is the naming - maybe we rather have this be a depthai base and the depthai for a fully fledged depthai image?

**Comment by VanDavv** [comment]
I would opt to use https://github.com/luxonis/depthai-docker repository for this

File: .github/workflows/docker-hub.yml (hunk 1 above)
Comment (pamolloy) [reply .github/workflows/docker-hub.yml:13]: Nope! It is something I'd like to implement at some point, but not worth adding here.

**Comment by pamolloy** [comment]
@VanDavv @themarpe The reason I added the Dockerfile here is because it uses `install_dependencies.sh`. I just added the condition (which I removed while debugging) that it only builds an image when `install_dependencies.sh` is modified. Hopefully that works in conjunction with `push: branches: master`. :crossed_fingers: 

Note that the installation instructions that document the use of `install_dependencies.sh` will move to `python-depthai` in Gen 2, so I assume `install_dependencies.sh` will also move to `python-depthai`. And naturally the `Dockerfile` would move too.

I don't think the Dockerfile needs it's own repository, not to mention the above point about keeping it close to `install_dependencies.sh`.

**Review by themarpe — APPROVED** [review]

**Pull request merged**